src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs

```diff
@@ -231,14 +233,15 @@ private async Task UpdateSourceSymbolTreeInfoAsync(Project project, Cancellation
             }
 
             [PerformanceSensitive("https://github.com/dotnet/roslyn/issues/36158", AllowCaptures = false, Constraint = "Avoid captures to reduce GC pressure when running in the host workspace.")]
-            private Task GetTask(Project project, Func<IncrementalAnalyzer, Project, PortableExecutableReference, CancellationToken, Task> func, PortableExecutableReference reference, CancellationToken cancellationToken)
+            private Task GetTask<TReference>(Project project, Func<IncrementalAnalyzer, Project, TReference, CancellationToken, Task> func, TReference reference, CancellationToken cancellationToken)
+                where TReference : MetadataReference?
```

Comment:
why `PortableExecutableReference` is changed to `MetadataReference`?

Reply:
The argument needs to get passed to the delegate with the same nullability (i.e. allow null for the argument, but if a non-null argument is passed then the delegate also sees a non-null argument). `PortableExecutableReference` is derived from `MetadataReference`, so the constraint was simplified.